File: Java/Solutions/8.1. Exam Preparation - Part I/src/PointOnSegment.java
import java.util.Scanner;


public class PointOnSegment {
    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        int first = Integer.parseInt(scanner.nextLine());
        int second = Integer.parseInt(scanner.nextLine());
        int point = Integer.parseInt(scanner.nextLine());

        int left = Math.min(first, second);
        int right = Math.max(first, second);

        int distanceLeft = Math.abs(left - point);
        int distanceRight = Math.abs(right - point);

        int minDistance = Math.min(distanceLeft, distanceRight);

        if (point >= left && point <= right){
            System.out.println("in");
        }
        else{
            System.out.println("out");
        }

        System.out.println(minDistance);
    }
}

File: JavaScript/Solutions/08.Exam Preparation - Part I/PointOnSegment.js
function solve([first, second, point]) {
    first = Number(first);
    second = Number(second);
    point = Number(point);

    let left = Math.min(first, second);
    let right = Math.max(first, second);

    let distanceLeft = Math.abs(left - point);
    let distanceRight = Math.abs(right - point);
    
    let minDistance = Math.min(distanceLeft, distanceRight);

    if(point >= left && point <= right) {
        console.log('in');
    }
    else {
        console.log('out');
    }

    console.log(minDistance);
}

solve([10, 5, 7]);
solve([8, 10, 5]);
solve([1, -2, 3]);